engine/src/main/java/org/terasology/recording/RecordedEventSerializer.java

```diff
@@ -0,0 +1,515 @@
+            ButtonState state = (ButtonState) typeHandler.deserialize(data, deserializationContext);
+            char keychar = jsonObject.get("keychar").getAsCharacter();
+            float delta = jsonObject.get("delta").getAsFloat();
+            KeyEvent aux;
+            if (c.equals(KeyDownEvent.class.toString())) {
+                aux = KeyDownEvent.create(input, keychar, delta);
+            } else if (c.equals(KeyRepeatEvent.class.toString())) {
+                aux = KeyRepeatEvent.create(input, keychar, delta);
+            } else {
+                aux = KeyUpEvent.create(input, keychar, delta);
+            }
+            aux.setState(state);
+            newEvent = aux;
+        } else if (c.equals(MouseButtonEvent.class.toString())) {
+            GsonPersistedData data = new GsonPersistedData(jsonObject.get("button"));
+            TypeHandler typeHandler = typeSerializationLibrary.getTypeHandlerFromClass(MouseInput.class);
+            MouseInput button = (MouseInput) typeHandler.deserialize(data, deserializationContext);
+            data = new GsonPersistedData(jsonObject.get("state"));
+            typeHandler = typeSerializationLibrary.getTypeHandlerFromClass(ButtonState.class);
+            ButtonState state = (ButtonState) typeHandler.deserialize(data, deserializationContext);
+            JsonObject aux = jsonObject.get("mousePosition").getAsJsonObject();
+            Vector2i mousePosition = new Vector2i(aux.get("x").getAsInt(), aux.get("y").getAsInt());
+            float delta = jsonObject.get("delta").getAsFloat();
+            MouseButtonEvent event = new MouseButtonEvent(button, state, delta);
+            event.setMousePosition(mousePosition);
+            newEvent = event;
+        } else if (c.equals(MouseAxisEvent.class.toString())) {
+            GsonPersistedData data = new GsonPersistedData(jsonObject.get("mouseAxis"));
+            TypeHandler typeHandler = typeSerializationLibrary.getTypeHandlerFromClass(MouseAxisEvent.MouseAxis.class);
+            MouseAxisEvent.MouseAxis mouseAxis = (MouseAxisEvent.MouseAxis) typeHandler.deserialize(data, deserializationContext);
+            float value = jsonObject.get("value").getAsFloat();
+            float delta = jsonObject.get("delta").getAsFloat();
+            newEvent = MouseAxisEvent.create(mouseAxis, value, delta);
+        } else if (c.equals(MouseWheelEvent.class.toString())) {
+            JsonObject aux = jsonObject.get("mousePosition").getAsJsonObject();
+            Vector2i mousePosition = new Vector2i(aux.get("x").getAsInt(), aux.get("y").getAsInt());
+            int wheelTurns = jsonObject.get("wheelTurns").getAsInt();
+            float delta = jsonObject.get("delta").getAsFloat();
+            newEvent = new MouseWheelEvent(mousePosition, wheelTurns, delta);
+        } else {
+            logger.error("Not an Input Event"); // change to logger
+            return null;
+        }
```

Comment:
same as above. default constructors can probably be simplified with reflection and class.newInstance.